[Tan Ming Ann] iP

skythefire opened the pull request:
### **Jo**

> “I got serious task to do aka clear _every_ bit of the mookata before I die in the 30 mins” – Darrion

Jo is a bot-wannabe frog that helps to keep track of all your pesky ~~flies~~ tasks you have on your ~~menu~~ to-do list.

Jo is:
- text-based
- easy to use
- hungry for the tasks that you have for her!

All you need to do is,
1. download it from [here](https://github.com/skythefire/ip/releases).
2. add your ~~tasks~~ flies.
3. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Searching ~~flies by flavor profile~~ tasks by keywords
- [ ] Reminders (coming soon)

Credit: Josephine & Darrion for being my source of inspiration for this IP :skateboard:

------------------------------------------------------------------------------------
Here's a preview of Jo greeting you!
```  
    _____
   /     \______
  | o     |     \____
  /\_____/           \___
 /                       \
|_______/                 \
  \______   _       ___    \
        /\_//      /   \    |
       // //______/    /___/
      /\/\/\      \   / \ \
                    \ \   \ \
                      \ \   \ \
                        \ \  \ \
                         \ \ /\/\
                         /\/\

public void greeting() {
    String greeting = "I am Jo the Frog! RIBBIT! \n";
    System.out.println(frog + greeting + "How may I help you?\n" + LINE);
}

Commit:
Add Gradle support

skythefire committed:
Added frog icon and greeting

skythefire committed:
Added Echo and Exit functions

skythefire committed:
Tidied code; added Add, List functions

skythefire committed:
Abstracted out store array into a Task class

skythefire committed:
Added Mark as Done feature

skythefire committed:
Added ToDo, Deadline and Event classes

skythefire committed:
Add support for tracking three types of tasks

skythefire committed:
Add I/O redirection

skythefire committed:
Handle some errors in Duke

skythefire committed:
Abstracted into an ArrayList

skythefire committed:
Added Delete function

skythefire committed:
Un-hardcoded reading of user inputs

skythefire committed:
Added Save function

skythefire committed:
Added Date and Time support for Deadline

skythefire committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Added Date and Time support for Deadline

skythefire committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Added Save function

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java

skythefire committed:
Added more OOP

skythefire committed:
Organised into duke package

skythefire committed:
Added JUnit Tests

skythefire committed:
Packaged as executable JAR file

skythefire committed:
Added JavaDocs

skythefire committed:
Touched up JavaDocs

skythefire committed:
Adopt coding standards

skythefire committed:
Add Find function

skythefire committed:
Merge branch 'branch-A-JavaDoc'

skythefire committed:
Merge conflicts

skythefire committed:
Merge conflicts

skythefire committed:
Merge remote-tracking branch 'origin/add-gradle-support'

skythefire committed:
Added GUI for Duke

skythefire committed:
Merge branch 'branch-Level-10'

skythefire committed:
Fix coding standards

leonghongfai reviewed (commented):
Looks good! 👍

asherhy commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,56 @@
+        this.time = time;
+    }
+
+    /**
+     * Gets the date the task must be completed by.
+     * @return the date the task must be completed by
+     */
+    public String getBy() {
+        return this.by.toString();
+    }
+
+    /**
+     * Gets the time the task must be completed by.
+     * @return the string representing the time the task must be completed by
+     */
+    public String getTime() {
+        return this.time;
+    }
```
I like how you have separated the time and date!

asherhy replied:
It is nice that you separate the time and date

asherhy commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,26 @@
+     * @param str the user input
+     * @throws DukeException If input does not match Duke's command words
+     */
+    public static void checkInput(String str) throws DukeException {
```
Maybe you can consider using a clearer name like String input

asherhy replied:
maybe you can change the argument to input as str might be too vague

asherhy commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,143 @@
+     * @param tasks The TaskList to be affected
+     */
+    public String addTaskMessage(TaskList tasks) {
+        return "A fly has been added to the menu:\n" +
```
code is clean

asherhy commented on src/main/java/duke/ToDo.java:
```diff
@@ -0,0 +1,26 @@
+    public String print() {
+        return String.format("T,%d,%s",isCompleted() ? 1 : 0, this.getName());
+    }
+}
```
code is clean and simple

asherhy reviewed (commented):
Overall code is simple and clean

skythefire committed:
Classes do not have assertions to check for

skythefire committed:
Merge pull request #1 from skythefire/branch-A-Assertions

Classes do not have assertions to check for

skythefire committed:
Merge branch 'master' of https://github.com/skythefire/ip

skythefire committed:
Update code quality

Refactor code to improve code quality
Extract out the code to make the icons a circle into its own method, and explain the
purpose of the code

skythefire committed:
Merge pull request #2 from skythefire/branch-A-CodeQuality

Update code quality

skythefire committed:
Merge branch 'master' of https://github.com/skythefire/ip

skythefire committed:
Add C-FriendlierSyntax

Make the command syntax more flexible.
Allows for easier addition of tasks to the task list.

For example:
Allows for usage of
- "t" instead of "todo"
- "d" instead of "deadline"
- "e" instead of "event"

skythefire committed:
Add UserGuide

skythefire committed:
Set theme jekyll-theme-minimal

skythefire committed:
Add a blank line before the table

skythefire committed:
Merge branch 'master' of https://github.com/skythefire/ip

skythefire committed:
Add markdown

skythefire committed:
Fix bugs

skythefire committed:
Bug fixes

skythefire committed:
Fix save to hard disk
